forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/apiclient/ApiClientIdentity.java

```diff
@@ -0,0 +1,156 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.aspsp.as.service.apiclient;
+
+import com.forgerock.openbanking.common.error.exception.oauth2.OAuth2InvalidClientException;
+import com.forgerock.openbanking.model.OBRIRole;
+import com.forgerock.spring.security.multiauth.model.authentication.X509Authentication;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.asn1.x500.RDN;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x500.style.BCStyle;
+import org.bouncycastle.asn1.x500.style.IETFUtils;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateHolder;
+import org.springframework.security.core.GrantedAuthority;
+
+import javax.security.auth.x500.X500Principal;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+
+@Slf4j
+public abstract class ApiClientIdentity {
+
+    X509Authentication authentication;
+
+    public ApiClientIdentity(X509Authentication certificateAuthentication) throws OAuth2InvalidClientException {
+        this.authentication = certificateAuthentication;
+        this.throwIfNotValidCertAuthority();
+    }
+
+    /**
+     * Returns true if the MATLS certificate presented by the ApiClient was a PSD2/eIDAS style certificate.
+     * @return true if an PSD2/eIDAS certificate, false otherwise
+     * */
+    public abstract boolean isPSD2Certificate();
+
+
+    /**
+     * Return the unique TppIdentifier. This can be overridden for different ApiClientIdentity types. For example,
+     * OBWacs should use the NCA registration ID found in the organisationIdentifier (oid 2.5.4.97) of the subject
+     * Name in the certificate. For an OB Transport (legacy OB Directory issued certs) we might need to use the OU
+     * field from the certificate subject....
+     * @return a UID for the TPP.
+     */
+    public String getTransportCertificateCn(){
+        try {
+            X509Certificate transportCert = getTransportCertificate();
+            if(transportCert != null) {
+                X500Name x500name = new JcaX509CertificateHolder(transportCert).getSubject();
+                RDN cn = x500name.getRDNs(BCStyle.CN)[0];
+                String cnString = IETFUtils.valueToString(cn.getFirst().getValue());
+                return cnString;
+            } else {
+                log.debug("getTppIdentifier() No certificates available from request");
```

Comment:
few places in the code like this. debug logs without any context. Going through the logs and seeing this would be frustrating as it tells us nothing about the request.

Reply:
Fixed